File: app/src/main/java/com/kazakago/cleanarchitecture/di/module/ApplicationModule.java
package com.kazakago.cleanarchitecture.di.module;

import android.content.Context;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;

@Module
public class ApplicationModule {

    private Context context;

    public ApplicationModule(Context context) {
        this.context = context.getApplicationContext();
    }

    @Provides
    @Singleton
    public Context provideApplicationContext() {
        return context;
    }

}

File: app/src/main/java/com/kazakago/cleanarchitecture/di/module/ApplicationModule.kt
package com.kazakago.cleanarchitecture.di.module

import android.content.Context
import dagger.Module
import dagger.Provides
import javax.inject.Singleton

@Module
class ApplicationModule(context: Context) {

    private val context: Context = context.applicationContext

    @Provides
    @Singleton
    fun provideApplicationContext(): Context {
        return context
    }

}